classification/classification_network.py

```diff
@@ -246,109 +220,368 @@ def createClassifierRegion(network, classifierType, classifierParams):
+  network
+  @param networkConfiguration (dict) the configuration of this network.
+  """
+
+  # create the network encoders for sensor data.  
+  minval, maxval = _findCSVMinMax(inputFile)
+  _setScalarEncoderMinMax(networkConfiguration, minval, maxval)
+  encoders = networkConfiguration[SENSOR_REGION_NAME]["encoders"]
+
+  # Create and run network on this data.
+  #   Input data comes from a CSV file (scalar values, labels). The
+  #   RecordSensor region allows us to specify a file record stream as the
+  #   input source via the dataSource attribute.
+  dataSource = FileRecordStream(streamID=inputFile)
+
+  network = createNetwork(dataSource,
+                          encoders,
+                          networkConfiguration)
+
+  # Need to init the network before it can run.
+  network.initialize()
   return network
+
+
+
+def _findNumberOfPartitions(networkConfiguration, numRecords):
+  """
+  Find the number of partitions for the input data based on a specific
+  networkConfiguration. 
+  
+  @param networkConfiguration (dict) the configuration of this network.
+  @param numRecords (int) Number of records of the input dataset.
+  @return partitions (list of tuples) Region names and associated indices 
+  partitioning the input dataset to indicate at which recordNumber it should 
+  start learning. The remaining of the data (last partition) is used as a test 
+  set. 
+  """
+
+  spEnabled = networkConfiguration[SP_REGION_NAME]["enabled"]
+  tmEnabled = networkConfiguration[TM_REGION_NAME]["enabled"]
+  upEnabled = networkConfiguration[UP_REGION_NAME]["enabled"]
+
+  partitionNames = []
+  if spEnabled and tmEnabled and upEnabled:
+    numPartitions = 5
+    partitionNames.extend([SP_REGION_NAME,
+                           TM_REGION_NAME,
+                           UP_REGION_NAME])
+  elif spEnabled and tmEnabled:
+    numPartitions = 4
+    partitionNames.extend([SP_REGION_NAME,
+                           TM_REGION_NAME])
+  elif spEnabled:
+    numPartitions = 3
+    partitionNames.append(SP_REGION_NAME)
+  else:
+    numPartitions = 2  # only the classifier, so just test and train partitions
+  partitionNames.append(CLASSIFIER_REGION_NAME)
+  partitionNames.append(TEST_PARTITION_NAME)
+
+  partitionIndices = [numRecords * i / numPartitions
+                      for i in range(0, numPartitions)]
+
+  return zip(partitionNames, partitionIndices)
+
+
+
+def _setScalarEncoderMinMax(networkConfiguration, minval, maxval):
+  """
+  Introspect an input file and set the min and max values.
+  
+  @param networkConfiguration (dict) the configuration of this network.
+  """
+  scalarEncoderParams = (networkConfiguration
+                         [SENSOR_REGION_NAME]
+                         ["encoders"]
+                         [SCALAR_ENCODER_NAME])
+  scalarEncoderParams["minval"] = minval
+  scalarEncoderParams["maxval"] = maxval
+
+
+
+def _findCSVMinMax(fileName):
```

Comment:
This is implemented for nupic record streams (see below). Did you try already?
- [FileRecordStream.getStats()](https://github.com/numenta/nupic/blob/master/nupic/data/file_record_stream.py#L494)
- [RecordStream.getFieldMin()](https://github.com/numenta/nupic/blob/master/nupic/data/record_stream.py#L365) and [getFieldMax()](https://github.com/numenta/nupic/blob/master/nupic/data/record_stream.py#L382)